src/main/java/gregtech/integration/theoneprobe/provider/CoverProvider.java

```diff
@@ -0,0 +1,233 @@
+        }
+    }
+
+    /**
+     * Displays text for {@link CoverConveyor} related covers
+     *
+     * @param probeInfo the info to add the text to
+     * @param conveyor  the conveyor to get data from
+     */
+    private static void conveyorInfo(@Nonnull IProbeInfo probeInfo, @Nonnull CoverConveyor conveyor) {
+        String rateUnit = " {*cover.conveyor.transfer_rate*}";
+
+        if (conveyor instanceof CoverItemVoiding) {
+            itemVoidingInfo(probeInfo, (CoverItemVoiding) conveyor);
+        }
+
+        // do not display the regular rate if the cover has a specialized rate
+        if (!(conveyor instanceof CoverItemVoiding) && (!(conveyor instanceof CoverRoboticArm) || ((CoverRoboticArm) conveyor).getTransferMode() == TransferMode.TRANSFER_ANY)) {
+            transferRateText(probeInfo, conveyor.getConveyorMode(), rateUnit, conveyor.getTransferRate());
+        }
+
+        ItemFilterContainer filter = conveyor.getItemFilterContainer();
+        if (conveyor instanceof CoverRoboticArm) {
+            CoverRoboticArm roboticArm = (CoverRoboticArm) conveyor;
+            transferModeText(probeInfo, roboticArm.getTransferMode(), rateUnit, filter.getTransferStackSize(), filter.getFilterWrapper().getItemFilter() != null);
+        }
+        itemFilterText(probeInfo, filter.getFilterWrapper().getItemFilter());
+    }
+
+    /**
+     * Displays info for {@link CoverItemVoiding} related covers
+     *
+     * @param probeInfo the info to add the text to
+     * @param voiding   the voiding cover to get data from
+     */
+    private static void itemVoidingInfo(@Nonnull IProbeInfo probeInfo, @Nonnull CoverItemVoiding voiding) {
+        String unit = " {*gregtech.top.unit.items*}";
+
+        ItemFilterContainer container = voiding.getItemFilterContainer();
+        if (voiding instanceof CoverItemVoidingAdvanced) {
+            CoverItemVoidingAdvanced advanced = (CoverItemVoidingAdvanced) voiding;
+            VoidingMode mode = advanced.getVoidingMode();
+            voidingText(probeInfo, mode, unit, container.getTransferStackSize());
+        }
+        itemFilterText(probeInfo, container.getFilterWrapper().getItemFilter());
+    }
+
+    /**
+     * Displays text for {@link CoverPump} related covers
+     *
+     * @param probeInfo the info to add the text to
+     * @param pump      the pump to get data from
+     */
+    private static void pumpInfo(@Nonnull IProbeInfo probeInfo, @Nonnull CoverPump pump) {
+        String rateUnit = IProbeInfo.STARTLOC + pump.getBucketMode().getName() + IProbeInfo.ENDLOC;
+
+        if (pump instanceof CoverFluidVoiding) {
+            fluidVoidingInfo(probeInfo, (CoverFluidVoiding) pump);
+        }
+
+        // do not display the regular rate if the cover has a specialized rate
+        if (!(pump instanceof CoverFluidVoiding) && (!(pump instanceof CoverFluidRegulator) || ((CoverFluidRegulator) pump).getTransferMode() == TransferMode.TRANSFER_ANY)) {
```

Comment:
Same here, this could just be an `else if` to avoid a check on CoverFluidVoiding